src/coreclr/vm/interpexec.cpp

```diff
@@ -1081,13 +1121,24 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
                     }
                     else
                     {
-                        // At this stage in the implementation, we assume this is pointer to
-                        // interpreter code. In the future, this should probably be tagged pointer
-                        // for interpreter call or normal pointer for JIT/R2R call.
                         targetIp = (const int32_t*)targetMethod;
                     }
                     }
 CALL_TARGET_IP:
+                    // Interpreter-TODO: we need a fast way to check of the targetIp is an interpreter code or not.
+                    // Probably use a tagged pointer for interpreter code and a normal pointer for JIT/R2R code.
+                    EECodeInfo codeInfo((PCODE)targetIp);
+                    if (!codeInfo.IsValid())
+                    {
+                        EEPOLICY_HANDLE_FATAL_ERROR_WITH_MESSAGE(COR_E_EXECUTIONENGINE, W("Attempted to execute native code from interpreter"));
+                    }
+                    else if (codeInfo.GetCodeManager() != ExecutionManager::GetInterpreterCodeManager())
```

Comment:
```suggestion
                    if (!codeInfo.IsValid() || codeInfo.GetCodeManager() != ExecutionManager::GetInterpreterCodeManager())
```
I do not see why we need to block native code (FCalls?) here. Should this be like this?

Reply:
This was initially a temporary check to make sure something unexpected doesn't leak in here. As the comment mentions, I want to get rid of the codeInfo construction here soon and move to tagged pointer for the interpreter code so that we don't need to burn time looking up the code ranges. Then this will go away.
As for FCalls, I have made a local change in a testing branch yesterday to make them work, but they need to be handled explicitly, as their GetNativeCode returns NULL, so I have used the TryGetMultiCallableAddrOfCode. Maybe we can use that function instead of the GetNativeCode for all calls, but I need to refresh my memory on the differences.